python/cudf_polars/cudf_polars/experimental/rapidsmpf/shuffle.py

```diff
@@ -169,7 +172,7 @@ def extract_chunk(self, sequence_number: int) -> plc.Table:
         partition_chunks = self.shuffler.extract(sequence_number)
         return py_unpack_and_concat(
             partitions=partition_chunks,
-            stream=self.stream,
+            stream=stream,
```

Comment:
So here, we have `partition_chunks`, which is a `list[PackedData]`. Each individual `PackedData` should be valid on some stream. In C++ that's `packed_data.data.stream` ([here](https://github.com/rapidsai/rapidsmpf/blob/decd6ca4d756f54cb9b57a7c874785894c4ba6a4/cpp/include/rapidsmpf/buffer/packed_data.hpp#L28)).

However, we don't appear to have access to that from Python. I think we'll need to fix that in rapidsmpf.

Then we can get a new CUDA stream that's downstream of each of those.

```
stream = get_joined_cuda_stream(ir_context.get_cuda_stream, [x.data.stream for x in partition_chunks])
```

Reply:
Turns out `unpack_and_concat` does all this for us: https://github.com/rapidsai/rapidsmpf/blob/decd6ca4d756f54cb9b57a7c874785894c4ba6a4/cpp/include/rapidsmpf/integrations/cudf/partition.hpp#L129-L131.

So we don't have to do any (additional) stream joining here.

Since the `stream` passed in here comes from `ir_context.get_cuda_stream()` should be good like you have it.